content/blog/automation-api-workflow/index.md

```diff
@@ -0,0 +1,350 @@
+
+"This example provisions an AWS Aurora SQL database and executes a database "migration" using the resulting connection info. This migration creates a table, inserts a few rows of data, and reads the data to verify the setup. This is all done in a single inline Pulumi program."
+
+We'll expand on this example and add the following:
+
+- show how to retrieve a password from AWS Secrets Manager
+- add functions for database backup, mysqlcheck, optimize, and analyze
+
+Let's get started by cloning the repository and open the `index.ts`.
+
+```bash
+$ git clone https://github.com/pulumi/automation-api-examples.git
+$ cd ./automation-api-examples/nodejs/databaseMigration-ts/
+```
+
+## Program overview
+
+TSeveral pieces comprise the program; the first is the program's main scope, where we can pass values from the functions that make up the script. The next part is the `pulumiProgram` function that declares the infrastructure.
+
+```typescript
+const run = async () => {
+    // This is our pulumi program in "inline function" form
+    const pulumiProgram = async () => {
+```
+
+To deploy the infrastructure, we use the Automation API to instantiate the resources and create a stack. Like a Pulumi program invoked by the CLI, we need to provide a stack and project name and the inline program's name that declares the infrastructure.
+
+```typescript
+    // Create our stack
+    const args: InlineProgramArgs = {
+        stackName: "dev",
+        projectName: "databaseMigration",
+        program: pulumiProgram
+    };
+
+    // create (or select if one already exists) a stack that uses our inline program
+    const stack = await LocalWorkspace.createOrSelectStack(args);
+```
+
+Once the infrastructure is created, we can call provisioning functions, such as `createdb`, maintenance functions, or other functions to manage the database.
+
+## Adding arguments for new commands
+
+First, we'll add the arguments for database creation, mysqlcheck, optimize, and analyze. These arguments call the functions that build the infrastructure, create and populate the database, and perform maintenance.
+
+```typescript
+const process = require('process');
+
+const args = process.argv.slice(2);
+let destroy = false;
+let createdb =false;
+let backup = false;
+let mysqlcheck = false;
+let optimize = false;
+let analyze = false;
+
+if (args.length > 0 && args[0]) {
+    destroy = args[0] === "destroy";
+    createdb = args[0] === "createdb";
+    backup = args[0] === "backup";
+    mysqlcheck  = args[0] ===  "mysqlcheck";
+    optimize  = args[0] === "optimize";
+    analyze = args[0] === "analyze";
+}
+```
+
+## Declare infrastructure
+
+The all the code used to create the infrastructure (a VPC, a security group, and an RDS, provision the database, and maintain the database is in an inline program, which is a function that we can call using [yarn](https://yarnpkg.com/) instead of the Pulumi CLI.
```

Comment:
> using [yarn](https://yarnpkg.com/) instead of the Pulumi CLI.

This should probably say:
> using Node.js directly instead of the Pulumi CLI

In this case, yarn is just the package manager that invokes the start script in the `package.json`:

https://github.com/pulumi/automation-api-examples/blob/main/nodejs/databaseMigration-ts/package.json#L18